ga4gh/avrotools.py

```diff
@@ -152,9 +223,194 @@ def _getInvalidFields(cls, expectedSchema, datum):
             for field in expectedSchema.fields:
                 key = field.name
                 value = datum.get(key)
-                result = cls._getInvalidFields(field.type, value)
-                if result != cls.sinkValue:
+                result = self._getInvalidFields(field.type, value)
+                if result != self.sinkValue:
                     dic[key] = result
             if len(dic):
                 return dic
-        return cls.sinkValue
+        return self.sinkValue
+
+
+class RandomSchemaTool(object):
```

Comment:
This sounds like a tool for generating random schemas to me. What about `RandomInstanceGenerator`? I'm a bit suspicious of anything ending in 'tool' --- it sounds a bit vague and generic... Likewise, I would prefer something like `InstanceValidator` rather than `SchemaTool` above.